test/unit/RelayServer.test.ts

```diff
@@ -390,68 +472,134 @@ describe('RelayServer tests', function () {
+                data: dataWhenTransferingZero,
+              },
+              relayData: {
+                gasPrice: GAS_PRICE,
+              },
+            },
+          } as EnvelopingTxRequest);
 
-      it('Should charge fees based on transfer value, when transferFeePercentage > 0 and a transferFrom() is being relayed', async function () {
-        const fakeServerConfigParams = {
-          app: {
-            gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
-            transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
-          },
-        };
-        sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
+          expect(maxPossibleGas.toString()).to.be.eq(
+            FAKE_ESTIMATION_BEFORE_FEES.toString()
+          );
+        });
 
-        const maxPossibleGas = await relayServer.getMaxPossibleGas({
-          relayRequest: {
-            request: {
-              tokenContract: constants.AddressZero,
-              tokenAmount: TOKEN_AMOUNT_IN_REQUEST,
-              data: dataWhenTransferFrom,
+        it('Should charge fees based on transfer value, when a transferFrom() is being relayed', async function () {
+          const fakeServerConfigParams = {
+            app: {
+              gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+              transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
             },
-            relayData: {
-              gasPrice: GAS_PRICE,
+          };
+          sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
+
+          const maxPossibleGas = await relayServer.getMaxPossibleGas({
+            relayRequest: {
+              request: {
+                tokenContract: constants.AddressZero,
+                tokenAmount: TOKEN_AMOUNT_IN_REQUEST,
+                data: dataWhenTransferFrom,
+              },
+              relayData: {
+                gasPrice: GAS_PRICE,
+              },
             },
-          },
-        } as EnvelopingTxRequest);
+          } as EnvelopingTxRequest);
 
-        expect(maxPossibleGas.toString()).to.be.eq(
-          BigNumberJs(fakeEstimationBeforeFees)
-            .plus(expectedFeeFromTransfer)
-            .toString()
-        );
+          expect(maxPossibleGas.toString()).to.be.eq(
+            BigNumberJs(FAKE_ESTIMATION_BEFORE_FEES)
+              .plus(expectedFeeFromTransfer)
+              .toString()
+          );
+        });
+
+        it('Should not charge fees if the amount to transfer is zero when a transferFrom() is being relayed', async function () {
+          const dataWhenTransferingZero = dataWhenTransferFrom.replace(
+            tokenAmountToTransferAsHex,
+            '0'.repeat(64)
+          );
+
+          const fakeServerConfigParams = {
+            app: {
+              gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+              transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
+            },
+          };
+          sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
```

Comment:
We could move this in the `beforeEach` section

Reply:
Done.